dask_ml/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,340 @@
+            X = check_array(X, copy=self.copy, dtype=[np.float64, np.float32],
+                            accept_multiple_blocks=True)
+        n_samples, n_features = X.shape
+        if not hasattr(self, 'components_'):
+            self.components_ = None
+
+        if self.n_components is None:
+            if self.components_ is None:
+                self.n_components_ = min(n_samples, n_features)
+            else:
+                self.n_components_ = self.components_.shape[0]
+        elif not 1 <= self.n_components <= n_features:
+            raise ValueError("n_components=%r invalid for n_features=%d, need "
+                             "more rows than columns for IncrementalPCA "
+                             "processing" % (self.n_components, n_features))
+        elif not self.n_components <= n_samples:
+            raise ValueError("n_components=%r must be less or equal to "
+                             "the batch number of samples "
+                             "%d." % (self.n_components, n_samples))
+        else:
+            self.n_components_ = self.n_components
+
+        if (self.components_ is not None) and (self.components_.shape[0] !=
+                                               self.n_components_):
+            raise ValueError("Number of input features has changed from %i "
+                             "to %i between calls to partial_fit! Try "
+                             "setting n_components to a fixed value." %
+                             (self.components_.shape[0], self.n_components_))
+
+        # This is the first partial_fit
+        if not hasattr(self, 'n_samples_seen_'):
+            self.n_samples_seen_ = 0
+            self.mean_ = .0
+            self.var_ = .0
+        
+        # Update stats - they are 0 if this is the first step
+        last_sample_count = np.tile(np.expand_dims(self.n_samples_seen_, 0), X.shape[1])
+        col_mean, col_var, n_total_samples = \
+            _incremental_mean_and_var(
+                X, last_mean=self.mean_, last_variance=self.var_,
+                last_sample_count=last_sample_count)
+        n_total_samples = n_total_samples[0]
+        
+        # Whitening
+        if self.n_samples_seen_ == 0:
+            # If it is the first step, simply whiten X
+            X -= col_mean
+        else:
+            col_batch_mean = np.mean(X, axis=0)
+            X -= col_batch_mean
+            # Build matrix of combined previous basis and new data
+            mean_correction = \
+                np.sqrt((self.n_samples_seen_ * n_samples) /
+                        n_total_samples) * (self.mean_ - col_batch_mean)
+            X = np.vstack((self.singular_values_.reshape((-1, 1)) *
+                           self.components_, X, mean_correction))
+
+        solver = self._get_solver(X, self.n_components_)
+        if solver in {"full", "tsqr"}:
+            if hasattr(X, 'rechunk'):
+                X = da.rechunk(X, (X.chunks[0], -1))
```

Comment:
If the array is chunked along the last axis, svd won't work.
I'm not sure if rechunking is a right way here.
Probably we can raise an Exception?

Reply:
Perhaps check what we do elsewhere in the library. It's not clear to me whether the estimator should implicitly rechunk or not, since this could cause surprisingly high memory usage.